### ISSUE TITLE
[Feature Request] Adhere to popular TwinView/Xinerama

### ISSUE THREAD

**Issue by Redsandro** [linuxmint/cinnamon-screensaver#205]
When using a DP 1.2 UHD/4k high resolution monitor like the **Dell Ultrasharp UP2414Q**, [MST](https://en.wikipedia.org/wiki/Multi-Stream_Transport) is used to transfer _two_ half screens to _one_ monitor in 60 FPS. 

E.g.: `DP-0.9: 1920x2160 +0+0, DP-0.8: 1920x2160 +1920+0` is total 3840x2160+0+0

Windows maximize to the monitor (with two screens) just fine, so that part of the system understands what's going on. However, for some reason, the screensaver and lock screen don't get this. So the screensaver is shown twice, side by side, and the login prompt is shown on one half only.

![image](https://cloud.githubusercontent.com/assets/1702193/22148476/99b17492-df0e-11e6-9a65-1a8a44f96b64.png)

The same problem occurs when using two or three (vertical) monitors as one display/desktop. Windows maximize fine, but the screensaver and lock screen doesn't stretch to the configured size.

In nVidia drivers this is called `TwinView`. It would be nice if the screensaver and lock screen could act accordingly. I'm aware that this is nVidia-specific so I don't know how AMD/Intel folks do it, but MST and TwinView are getting more and more common. Windows does this out of the box (Win + P) and OSX does this automatically. Linux users need to be professionals to get this to work unfortunately, but some day it will be just as easy.

---

Possibly related errors:

* https://github.com/linuxmint/cinnamon-control-center/issues/142
* https://github.com/linuxmint/Cinnamon/issues/5385
* https://github.com/linuxmint/Cinnamon/issues/5654

**Comment by mtwebster** [comment]
May be fixed by https://github.com/linuxmint/cinnamon-screensaver/commit/c3715694aba68423bcde67b96b10462b41469939, but I have no way of testing this configuration currently.

**Comment by Redsandro** [comment]
@mtwebster once this will be available through normal channels I will test it.

**Comment by mtwebster** [comment]
I'm not sure how complicated this configuration is to set up, but if you're interested you could try this on a livecd session:

Run "Software Sources from the menu,
Check "Enable source code repositories"
Click "Update the cache"

then, in a terminal...
```
sudo apt-get build-dep cinnamon-screensaver
sudo apt-get install git

git clone https://github.com/linuxmint/cinnamon-screensaver.git

cd cinnamon-screensaver
dpkg-buildpackage
cd ..
sudo dpkg -i *.deb

cinnamon-screensaver-command -e                               ...(kills old screensaver)
cinnamon-screensaver
```

If you need the nvidia drivers then this probably isn't feasible.  The same steps would work on a regular install as well.  If you wanted to go back to the release version, you can get it back via the update manager.

Regardless, I will try to get an update out asap.

**Comment by Redsandro** [comment]
Yes, I need the nvidia drivers for the setup. I'm still running Linux Mint 18 Sarah. Will the update be backported or do I need to upgrade to 18.1?

Can I build/test the package from source on 18.0, or would that need 18.1 too?

**Comment by Redsandro** [comment]
@mtwebster I'm guessing I need to upgrade to 18.1 before attempting again.

`dpkg-checkbuilddeps: error: Unmet build dependencies: devscripts dh-exec librsvg2-bin python-rsvg libgirepository1.0-dev gobject-introspection`

Otherwise the `apt build-dep cinnamon-screensaver` would have taken care of this, no?
